File: BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsEventDao.java
package com.braintreepayments.api;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
interface AnalyticsEventDao {

    @Insert
    void insertEvent(AnalyticsEvent event);

    @Query("SELECT * FROM analytics_event")
    List<AnalyticsEvent> getAllEvents();

    @Delete
    void deleteEvents(List<AnalyticsEvent> events);
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsEventDao.kt
package com.braintreepayments.api

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
internal interface AnalyticsEventDao {

    @Insert
    fun insertEvent(event: AnalyticsEvent)

    @Query("SELECT * FROM analytics_event")
    fun getAllEvents(): List<AnalyticsEvent>

    @Delete
    fun deleteEvents(events: List<AnalyticsEvent>)
}
